_posts/2024-02-01-testing-with-wildfly-glow.adoc

```diff
@@ -0,0 +1,180 @@
+
+== Testing using the WildFly Maven Plugin to provision a server
+
+That is the simplest approach that applies to a single application deployed into WildFly. The WildFly quickstarts 
+cover this approach with the `provisioned-server` profile. Each quickstart defines this profile as you can find in the quickstarts https://github.com/wildfly/quickstart[source code].
+
+The `package` goal of the https://github.com/wildfly/wildfly-maven-plugin[WildFly Maven Plugin] scans the application, 
+discovers the provisioning configuration then provisions the server and deploy the application. 
+The server is started (`wildfly:start` goal) then the tests are executed to interact with the application deployed in the server.
+
+In some cases you need to run tests from inside the server or you want to tests multiple deployments, 
+that is where the WildFly Arquillian Adapter is needed.
+
+== Testing using the WildFly Arquillian Adapter
+
+=== Arquillian deployments
+
+When using Arquillian, the deployments to be deployed to the WildFly server are created by static Java methods 
+annotated with the `@org.jboss.arquillian.container.test.api.Deployment` annotation. 
+These deployments are created and deployed during test execution by Arquillian.
+
+=== WildFly Glow Arquillian Maven Plugin
+
+Details on how to use and configure the plugin can be found in the https://docs.wildfly.org/wildfly-glow/#_wildfly_glow_arquillian_maven_plugin[plugin documentation], 
+in this blog post we are introducing its usage.
+
+The WildFly Glow Arquillian Maven plugin is scanning all the Arquillian deployments (prior to the test execution) that are returned by the static methods 
+to discover the required Galleon layers.
+
+A provisioning.xml file containing the discovered layers and feature-packs is generated. 
+This Galleon configuration file is then used by the actual provisioning tooling (https://docs.wildfly.org/galleon/[Galleon Maven Plugin], 
+https://github.com/wildfly/wildfly-maven-plugin[WildFly Maven Plugin], 
+https://github.com/wildfly-extras/wildfly-jar-maven-plugin[WildFly JAR Maven Plugin]) 
+to provision the test server.
+
+In order to validate that what the scanning has discovered is what you are expecting, you can configure the `wildfly-glow-arquillian-plugin`
+maven plugin to contain the element `<expected-discovery>`.
+
+For example:
+
+----
+<expected-discovery>[cdi, ee-integration, ejb, ejb-lite, elytron-oidc-client, naming, servlet]==>ee-core-profile-server,ejb,elytron-oidc-client</expected-discovery>
+----
+
+The left part of the arrow contains the list of the discovered layers according to the scanning. 
+The right part is what will get provisioned. Composed of a base layer (always `ee-core-profile-server`) and 
+a list of the discovered layers not contained in the base layer.
+
+During plugin execution, the discovered layers are printed. If that is valid with respect to your tests, 
```

Comment:
Maybe also point out that the format of the expected-discovery string is the same as when running the CLI with -Dcompact (or whatever it is called)

Reply:
I am not sure, we are not exposing the -Dcompact from wildfly-glow CLI. You need to call CLI jar file directly to set system properties. It is more a for Glow own testing I would say.